File: [0232][Easy][Implement-queue-using-stacks]/Implement-queue-using-stacks.cpp
class MyQueue
{
public:
    /** Initialize your data structure here. */
    stack<int> a;
    stack<int> b;
    MyQueue()
    {
    }

    /** Push element x to the back of queue. */
    void push(int x)
    {
        a.push(x);
    }

    /** Removes the element from in front of queue and returns that element. */
    int pop()
    {
        int len;
        len = a.size();
        for (int i = 0; i < len; ++i)
        {
            b.push(a.top());
            a.pop();
        }
        int cur;
        cur = b.top();
        b.pop();
        for (int j = 0; j < len - 1; ++j)
        {
            a.push(b.top());
            b.pop();
        }
        return cur;
    }

    /** Get the front element. */
    int peek()
    {
        int len;
        len = a.size();
        for (int i = 0; i < len; ++i)
        {
            b.push(a.top());
            a.pop();
        }
        int cur;
        cur = b.top();
        for (int j = 0; j < len; ++j)
        {
            a.push(b.top());
            b.pop();
        }
        return cur;
    }

    /** Returns whether the queue is empty. */
    bool empty()
    {
        return a.empty() && b.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: [0232][Easy][Implement-queue-using-stacks]/Implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.stackin = []
        self.stackout = []

    def push(self, x: int):
        """
        Push element x to the back of queue.
        """
        self.stackin.append(x)

    def pop(self):
        """
        Removes the element from in front of queue and returns that element.
        """
        if not self.stackout:
            while self.stackin:
                a = self.stackin.pop()
                self.stackout.append(a)
        return self.stackout.pop()

    def peek(self):
        """
        Get the front element.
        """
        if not self.stackout:
            while self.stackin:
                a = self.stackin.pop()
                self.stackout.append(a)
        return self.stackout[-1]

    def empty(self):
        """
        Returns whether the queue is empty.
        """
        if not self.stackin and not self.stackout:
            return True
        else:
            return False


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()
